code/modules/mob/living/carbon/xenomorph/abilities/general_powers.dm

```diff
@@ -857,6 +857,11 @@
 		if(path_turf.density)
 			to_chat(stabbing_xeno, SPAN_WARNING("There's something blocking your strike!"))
 			return FALSE
+		for(var/obj/path_contents as obj in path_turf.contents)
+			if(path_contents.density && !path_contents.throwpass)
```

Comment:
not so sure about this throwpass check, is there a better way to check if it blocks `(PASS_MOB_THRU_XENO|PASS_OVER_THROW_MOB|PASS_TYPE_CRAWLER)` specifically?

Reply:
I don't know. I can't tell why the barrier code isn't working. The only thing I could get to work reliably is that combination.

Reply:
hmm, in that case maybe filter out stuff that's ONBORDER like barricades (check the flag) so you can at least still stab over those in accordance to the other bits of code